Dockerfile

```diff
@@ -1,92 +1,63 @@
-FROM ubuntu:20.04
-
-ENV TZ Europe/London
-
-RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-
-RUN echo "deb http://security.ubuntu.com/ubuntu bionic-security main" >> /etc/apt/sources.list
-
-# Install all the requirements
-RUN apt-get update && apt-get install -y \
+FROM ruby:3.1.4-alpine3.19 AS base
```

Comment:
There is a more minimal version of Ubuntu we could use that has ruby pre-installed. I could make use of that.

Then, for Alpine I can make separate dockerfile and users can choose to build with that. The builds shouldn't conflict since the other containers would be the same. 

I'm pretty sure it can actually be the same Dockerfile and a user can add an additional argument to build the Alpine container. I'm guessing the default should be Ubuntu?

Reply:
> I'm guessing the default should be Ubuntu?

Yes, please.

And Ruby doesn't need to be pre-installed. In production we use rbenv to install and update Ruby. And while you are developing, we may update Ruby again one day and if you have rbenv then `./bin/setup` will install the right version for you. There's value in using the same setup to stay in sync.

Reply:
I guess I usually go without rbenv in other docker setups because it adds to the bulk. If you nuke and rebuild after every ruby upgrade you can keep the images compact and use up less space on your development PC. It keeps things tidy.

However, I'll just try to keep the Ubuntu setup relatively the same (with rbenv), and more advanced users can make use of the slimmer Alpine.

Reply:
By the way, is the current version of Ubuntu still 22 on production servers?

Reply:
No, it's Ubuntu 20. We had trouble updating to newer versions with our current version of Ansible.